--- a/js/Client.aspx	2015-01-31 22:50:04.843207973 +0100
+++ b/js/Client.aspx	2015-01-31 22:49:47.656547886 +0100
@@ -4,14 +4,14 @@
 <body>
 
 <%
-string [] files = System.IO.Directory.GetFiles("../client", "*.exe");
+string [] files = System.IO.Directory.GetFiles("/var/www/openpetra/client", "*.exe");
 
 foreach (string file in files)
 {
         Response.Write("<a href=\"" + System.IO.Path.GetFileName(file) + "\">" + System.IO.Path.GetFileName(file) +"</a> " + System.IO.File.GetLastWriteTime(file).ToString() + " " + new System.IO.FileInfo(file).Length.ToString() + " bytes<br/>\n");
 }
 
-files = System.IO.Directory.GetFiles("../client", "*.zip");
+files = System.IO.Directory.GetFiles("/var/www/openpetra/client", "*.zip");
 
 foreach (string file in files)
 {
