--- a/csharp/ICT/Petra/Server/lib/MSysMan/UserManager.cs	2014-11-26 14:40:05.019141575 +0000
+++ b/csharp/ICT/Petra/Server/lib/MSysMan/UserManager.cs	2014-11-26 14:41:55.699138208 +0000
@@ -352,7 +352,7 @@
                 PetraPrincipal.ProcessID = AProcessID;
                 AProcessID = 0;
 
-                if (UserDR.PasswordNeedsChange)
+                if (UserDR.PasswordNeedsChange && AUserID != "DEMO" && AUserID != "SYSADMIN")
                 {
                     // The user needs to change their password before they can use OpenPetra
                     PetraPrincipal.LoginMessage = SharedConstants.LOGINMUSTCHANGEPASSWORD;
