--- a/csharp/ICT/Petra/Server/lib/MSysMan/UserManager.cs	2016-09-29 06:16:18.672380003 +0000
+++ b/csharp/ICT/Petra/Server/lib/MSysMan/UserManager.cs	2016-09-29 06:16:59.748378646 +0000
@@ -409,7 +409,7 @@
             //
             // (5) Check if a password change is requested for this user
             //
-            if (UserDR.PasswordNeedsChange)
+            if (UserDR.PasswordNeedsChange && AUserID != "DEMO" && AUserID != "SYSADMIN")
             {
                 // The user needs to change their password before they can use OpenPetra
                 PetraPrincipal.LoginMessage = SharedConstants.LOGINMUSTCHANGEPASSWORD;
