diff --git a/csharp/ICT/Petra/Server/lib/MFinance/Gift/Gift.Importing.cs b/csharp/ICT/Petra/Server/lib/MFinance/Gift/Gift.Importing.cs
index d483c95..bb01f35 100644
--- a/csharp/ICT/Petra/Server/lib/MFinance/Gift/Gift.Importing.cs
+++ b/csharp/ICT/Petra/Server/lib/MFinance/Gift/Gift.Importing.cs
@@ -1306,7 +1306,7 @@ namespace Ict.Petra.Server.MFinance.Gift
             // Find the default Tax deductabilty from the motivation detail. This ensures that the column can be missing.
             AMotivationDetailRow motivationDetailRow = (AMotivationDetailRow)AValidationMotivationDetailTable.Rows.Find(
                 new object[] { FLedgerNumber, AGiftDetails.MotivationGroupCode, AGiftDetails.MotivationDetailCode });
-            string defaultTaxDeductible = ((motivationDetailRow != null) && motivationDetailRow.TaxDeductible) ? "yes" : "no";
+            string defaultTaxDeductible = ((motivationDetailRow != null) && !motivationDetailRow.IsTaxDeductibleAccountNull() && motivationDetailRow.TaxDeductible) ? "yes" : "no";
 
             AGiftDetails.TaxDeductible = ImportBoolean(Catalog.GetString("Tax deductible"),
                 FMainDS.AGiftDetail.ColumnTaxDeductible, AValidationControlsDictGiftDetail, defaultTaxDeductible);

