--- a/csharp/ICT/Petra/Server/lib/MPartner/connect/Partner.PartnerEdit.cs	2014-11-24 17:08:11 +0000
+++ b/csharp/ICT/Petra/Server/lib/MPartner/connect/Partner.PartnerEdit.cs	2014-11-27 13:38:12 +0000
@@ -3051,11 +3051,10 @@
         {
             TDBTransaction ReadTransaction;
             Boolean NewTransaction = false;
-            PartnerEditTDSPPartnerLocationTable LocationDT;
+            PartnerEditTDS PartnerLocations = new PartnerEditTDS();
             PLocationTable LocationTable;
             PLocationRow LocationRow;
 
-            LocationDT = new PartnerEditTDSPPartnerLocationTable();
             try
             {
                 ReadTransaction = DBAccess.GDBAccessObj.GetNewOrExistingTransaction(IsolationLevel.RepeatableRead,
@@ -3073,15 +3072,15 @@
                     {
                         ALocations = PLocationAccess.LoadViaPPartner(FPartnerKey, ReadTransaction);
 
-                        LocationDT.Merge(PPartnerLocationAccess.LoadViaPPartner(FPartnerKey, ReadTransaction));
+                        PPartnerLocationAccess.LoadViaPPartner(PartnerLocations, FPartnerKey, ReadTransaction);
 
-                        foreach (PartnerEditTDSPPartnerLocationRow row in LocationDT.Rows)
+                        foreach (PartnerEditTDSPPartnerLocationRow row in PartnerLocations.PPartnerLocation.Rows)
                         {
                             LocationTable = PLocationAccess.LoadByPrimaryKey(row.SiteKey, row.LocationKey, ReadTransaction);
 
                             if (LocationTable.Count > 0)
                             {
-                                LocationRow = (PLocationRow)LocationTable.Rows[0];
+                                LocationRow = LocationTable[0];
 
                                 row.LocationLocality = LocationRow.Locality;
                                 row.LocationStreetName = LocationRow.StreetName;
@@ -3095,14 +3094,14 @@
 
                                 if (!LocationRow.IsDateCreatedNull())
                                 {
-                                    row.LocationDateCreated = (DateTime)LocationRow.DateCreated;
+                                    row.LocationDateCreated = LocationRow.DateCreated.Value;
                                 }
 
                                 row.LocationModifiedBy = LocationRow.ModifiedBy;
 
                                 if (!LocationRow.IsDateModifiedNull())
                                 {
-                                    row.LocationDateModified = (DateTime)LocationRow.DateModified;
+                                    row.LocationDateModified = LocationRow.DateModified.Value;
                                 }
                             }
                         }
@@ -3111,7 +3110,7 @@
                     {
                         throw;
                     }
-                    ACount = LocationDT.Rows.Count;
+                    ACount = PartnerLocations.PPartnerLocation.Rows.Count;
                 }
             }
             finally
@@ -3122,7 +3121,7 @@
                     TLogging.LogAtLevel(7, "TPartnerEditUIConnector.GetPartnerLocationsInternal: committed own transaction.");
                 }
             }
-            return LocationDT;
+            return PartnerLocations.PPartnerLocation;
         }
 
         private PDataLabelValuePartnerTable GetDataLocalPartnerDataValuesInternal(out Boolean ALabelsAvailable, Boolean ACountOnly)

