diff -uNr OpenPetraNow-Release2014_05/setup/petra0300/linuxserver/postgresql/centos/openpetraorg-server.sh OpenPetraNow-Release2014_05_patched/setup/petra0300/linuxserver/postgresql/centos/openpetraorg-server.sh
--- OpenPetraNow-Release2014_05/setup/petra0300/linuxserver/postgresql/centos/openpetraorg-server.sh	2014-06-21 08:10:05.000000000 +0200
+++ OpenPetraNow-Release2014_05_patched/setup/petra0300/linuxserver/postgresql/centos/openpetraorg-server.sh	2014-06-24 15:41:41.235171000 +0200
@@ -25,7 +25,7 @@
   export OPENPETRA_PORT=@HostedPort@
   export POSTGRESQLVERSION=@PostgreSQL.Version@
   export backupfile=$OpenPetraOrgPath/backup30/backup-`date +%Y%m%d`.sql.gz
-  export mono_path=/opt/mono-openpetra
+  export mono_path=/opt/mono
   export FASTCGI_MONO_SERVER=fastcgi-mono-server4
   export mono=mono
 fi
