--- a/csharp/ICT/Common/CustomAppSettings.cs	2014-11-24 16:05:54.548242593 +0000
+++ b/csharp/ICT/Custom/CustomAppSettings.cs	2014-11-24 16:07:15.348240135 +0000
@@ -303,6 +303,15 @@
                 ReturnValue = ReturnValue.Replace("{applicationbindir}", ApplicationDirectory);
             }
 
+            if (AKey == "OpenPetra.HTTPServer" || AKey == "OpenPetra.Path.RemotePatches")
+            {
+            	string [] NameOfInstance = Directory.GetDirectories(TAppSettingsManager.ApplicationDirectory + "/../patches30");
+            	if (NameOfInstance.Length == 1)
+            	{
+            		ReturnValue = ReturnValue.Replace("demo", Path.GetFileName(NameOfInstance[0].ToLower()));
+            	}
+            }
+
             return ReturnValue;
         }
 
--- a/setup/petra0300/remoteclientWin/remote.nsi	2014-11-24 15:41:09.088287790 +0000
+++ b/setup/petra0300/remoteclientWin/remote.nsi	2014-11-24 15:51:15.220269348 +0000
@@ -7,6 +7,8 @@
 
   !include "MUI2.nsh"
   !include LogicLib.nsh
+  !include "StrFunc.nsh"
+  ${StrTok}
 ;--------------------------------
 ;General
 
@@ -88,6 +90,8 @@
   CreateDirectory "$INSTDIR\tmp30"
   CreateDirectory "$INSTDIR\patches30"
   CreateDirectory "$INSTDIR\bin30"
+  ${StrTok} $0 $EXEFILE "-" "1" "1"
+  CreateDirectory "$INSTDIR\patches30\$0"
   SetOutPath "$INSTDIR\bin30"
   File ..\..\..\csharp\ThirdParty\DevAge\SourceGrid.dll
   File ..\..\..\csharp\ThirdParty\SQLite\Mono.Data.Sqlite.dll

--- a/setup/petra0300/remoteclientWin/PetraClientRemote.config	2014-11-24 16:14:08.044227578 +0000
+++ b/setup/petra0300/remoteclientWin/PetraClientRemote.config	2014-11-24 16:16:18.648223604 +0000
@@ -6,8 +6,8 @@
     <add key="Reporting.PathReportSettings" value="{applicationbindir}\..\reports30\Settings" />
     <add key="Reporting.PathReportUserSettings" value="{applicationbindir}\..\reports30\Settings" />
     <add key="OpenPetra.Path.Patches" value="{applicationbindir}\..\patches30"/>
-    <add key="OpenPetra.Path.RemotePatches" value="http://${PublisherUrl}/patches/${OrgNameWithoutSpace}/"/>
-    <add key="OpenPetra.HTTPServer" value="http://${HostedServer}/openpetra${HostedPort}"/>
+    <add key="OpenPetra.Path.RemotePatches" value="https://demo.solidcharity.com/client/"/>
+    <add key="OpenPetra.HTTPServer" value="https://demo.solidcharity.com/"/>
     <add key="UINavigation.File" value="UINavigation.yml" />
     <add key="Resource.Dir" value="..\resources30" />
     <add key="UseWindowsUserID" value="true" />                <!-- Default Value: "false" -->

--- a/js/Default.aspx	2014-11-24 16:20:09.424216582 +0000
+++ b/js/Default.aspx	2014-11-24 17:09:21.220126769 +0000
@@ -31,12 +31,25 @@
     <div class="container">
 
       <form class="form-signin">
-        <h2 class="form-signin-heading">Please sign in to OpenPetra.js</h2>
+        <h2 class="form-signin-heading">Sign in to OpenPetra.js</h2>
         <input type="text" class="form-control" placeholder="Email address" autofocus id="txtEmail">
         <input type="password" class="form-control" placeholder="Password" id="txtPassword">
         <button id="btnLogin" class="btn btn-lg btn-primary btn-block" type="submit">Sign in</button>
+<% if (HttpContext.Current.Request.ServerVariables["HTTP_HOST"].Contains("demo")) { %>
+<div class="bs-callout bs-callout-warning"><h4>Login information</h4><p>Please login with user <b>demo</b> and password <b>demo</b>, or even easier, just click "Sign in"!</p></div>
+<% }
+string customer = HttpContext.Current.Request.ServerVariables["HTTP_HOST"];
+customer = customer.Substring(0, customer.IndexOf('.'));
+string [] files = System.IO.Directory.GetFiles("../client", "OpenPetraRemoteSetup-" + customer + "-*.exe");
+string filename = String.Empty;
+if (files.Length > 0)
+{
+   filename = files[0];
+}
+%>
+<div class="bs-callout bs-callout-warning"><h4>Download</h4><p>Get the <a href="https://<% Response.Write(HttpContext.Current.Request.ServerVariables["HTTP_HOST"]); %>/client/<% Response.Write(filename); %>">Windows Client</a>!</p></div>
+<div class="bs-callout bs-callout-info"><h4>You are welcome!</h4><p>Brought to you by <a href="https://www.solidcharity.com">www.solidcharity.com</a></p></div>
       </form>
-
     </div> <!-- /container -->
 
   </body>
