--- a/csharp/ICT/Petra/Client/app/Core/Cache.cs
+++ b/csharp/ICT/Petra/Client/app/Core/Cache.cs
@@ -1959,6 +1959,8 @@ namespace Ict.Petra.Client.App.Core
         /// </returns>
         public static DataTable LoadCacheableDataTableFromFile(String ATableName)
         {
+// see https://github.com/openpetra/openpetra/issues/63
+return null;
             DataTable ReturnValue;
             StreamReader CacheDTStreamReader;
             DataTable BinaryDT;
