--- a/setup/setup.build	2016-09-29 06:05:52.252400687 +0000
+++ b/setup/setup.build	2016-09-29 06:06:44.200398971 +0000
@@ -639,6 +639,38 @@
   </if>
 </target> 
 
+<target name="buildDemoSolidCharityCom">
+  <property name="WithLinuxClient" value="false"/>
+  <property name="WithWindowsClient" value="true"/>
+  <property name="OpenBuildService" value="true"/>
+  <property name="LinuxTargetDistribution-list" value="centos-mysql"/>
+  <property name="HostedPort" value="9000"/>
+  <property name="OrgName" value="by SolidCharity.com" overwrite="true"/>
+  <property name="OrgNameWithoutSpace" value="Demo" overwrite="true"/>
+  <property name="HostedServer" value="demo.solidcharity.com" overwrite="true"/>
+  <property name="PublisherUrl" value="demo.solidcharity.com" overwrite="true"/>
+  <property name="RemotePatchesURL" value="https://demo.solidcharity.com/client/" overwrite="true"/>
+  <property name="HostedServerURL" value="https://demo.solidcharity.com" overwrite="true"/>
+
+  <call target="nanttasks"/>
+  <call target="minimalGenerateSolution"/>
+  <call target="quickCompile"/>
+  <property name="DBMS.Type" value="mysql"/>
+  <call target="createSQLStatements"/>
+  <call target="translation"/>
+
+  <!-- build Remote Setup Windows installer -->
+  <property name="BuildArchive" value="client" overwrite="true"/>
+  <call target="prepareSetupDirForTarFile"/>
+  <call target="setupRemote"/>
+
+  <property name="BuildArchive" value="server" overwrite="true"/>
+  <call target="prepareSetupDirForTarFile"/>
+  <move file="${dir.setup}/centos/openpetra-server.sh" tofile="${dir.setup}/openpetra-server.sh"/>
+
+  <!-- result is in dir.setup for building the rpm package -->
+</target>
+
 <target name="packageReports">
        <exec program="${dir.delivery}/bin/Ict.Tools.PackageReports.exe" commandline="&quot;${dir.root}/XmlReports&quot; &quot;${dir.delivery}&quot;" />
 </target>
