--- openpetra-master/inc/nant/OpenPetra.default.targets.xml.orig	2014-12-24 08:14:25.776893951 +0000
+++ openpetra-master/inc/nant/OpenPetra.default.targets.xml	2014-12-24 08:14:36.784034758 +0000
@@ -181,6 +181,9 @@
     </if>
   </if>
   <if test="${'unix'==platform::get-name()}" >
+      <delete file="${dir.bin}/Mono.Security.dll" failonerror="false"/>
+      <delete file="${dir.bin}/Mono.Data.Sqlite.dll" failonerror="false"/>
+
       <!-- create symbolic links for OpenPetra.js -->
       <if test="${not directory::exists(dir.delivery + '/ThirdParty')}">
         <exec program="bash"
@@ -192,7 +195,7 @@
       <exec program="xsp4" 
               workingdir="${dir.delivery}"
               commandline="--verbose --nonstop --root . --port ${Server.Port} --applications /:."
-              spawn="false">
+              spawn="true">
         <environment>
           <variable name="PATH" path="${environment::get-variable('PATH')};${MonoBinPath}"/>
           <variable name="MONO_PATH" path="${MonoBinPath}\.."/>
--- openpetra-master/OpenPetra.build.orig       2014-12-23 22:09:49.013435756 +0000
+++ openpetra-master/OpenPetra.build    2014-12-24 06:51:44.305187540 +0000
@@ -547,8 +547,16 @@
-    <delete file="demoWith2ledgers.yml.gz" />
+    <delete file="demoWith1ledger.yml.gz" />
-    <exec program="wget" commandline="http://sourceforge.net/projects/openpetraorg/files/openpetraorg/demodata/demoWith2ledgers.yml.gz/download -OdemoWith2ledgers.yml.gz"/>
+    <exec program="wget" commandline="http://sourceforge.net/projects/openpetraorg/files/openpetraorg/demodata/demoWith1ledger.yml.gz/download -OdemoWith1ledger.yml.gz"/>
+    <loadtasks assembly="${Ict.Tools.NAntTasks.DLL}" unless="${task::exists('GenerateNamespaceMap')}"/>
+    <call target="initConfigFiles"/>
     <call target="startPetraServer"/>
+    <exec program="ps" commandline="xaf"/>
+    <exec program="netstat" commandline="-atplou"/>
+    <sleep seconds="2"/>
+    <exec program="ps" commandline="xaf"/>
+    <exec program="netstat" commandline="-atplou"/>
-    <sleep seconds="3"/>
+    <sleep seconds="15"/>
+    <exec program="ps" commandline="xaf"/>
+    <exec program="netstat" commandline="-atplou"/>
-    <loadtasks assembly="${Ict.Tools.NAntTasks.DLL}" unless="${task::exists('GenerateNamespaceMap')}"/>
     <ExecDotNet program="${dir.bin}/PetraServerAdminConsole.exe"
-        commandline="-C:${dir.root}/etc/ServerAdmin.config -Command:LoadYmlGz -YmlGzFile:${dir.root}/demoWith2ledgers.yml.gz"/>
+        commandline="-C:${dir.root}/etc/ServerAdmin.config -Command:LoadYmlGz -YmlGzFile:${dir.root}/demoWith1ledger.yml.gz"/>
     <call target="stopPetraServer"/>
--- openpetra-master/csharp/ICT/Petra/ServerAdmin/PetraServerAdminConsole/Main.cs.orig	2014-12-28 15:43:37.988756464 +0000
+++ openpetra-master/csharp/ICT/Petra/ServerAdmin/PetraServerAdminConsole/Main.cs	2014-12-28 15:44:16.773715624 +0000
@@ -789,6 +789,7 @@
                 Console.WriteLine("Exception occured while connecting/communicating to PETRAServer: " + exp.ToString());
             }
 
+            System.Environment.Exit(-1);
             return;
         }
 
