--- openpetra-master/OpenPetra.build.orig       2014-12-23 22:09:49.013435756 +0000
+++ openpetra-master/OpenPetra.build    2014-12-24 06:51:44.305187540 +0000
@@ -547,8 +547,9 @@
     <delete file="demoWith2ledgers.yml.gz" />
     <exec program="wget" commandline="http://sourceforge.net/projects/openpetraorg/files/openpetraorg/demodata/demoWith2ledgers.yml.gz/download -OdemoWith2ledgers.yml.gz"/>
     <call target="startPetraServer"/>
-    <sleep seconds="3"/>
+    <sleep seconds="15"/>
     <loadtasks assembly="${Ict.Tools.NAntTasks.DLL}" unless="${task::exists('GenerateNamespaceMap')}"/>
+    <call target="initConfigFiles"/>
     <ExecDotNet program="${dir.bin}/PetraServerAdminConsole.exe"
         commandline="-C:${dir.root}/etc/ServerAdmin.config -Command:LoadYmlGz -YmlGzFile:${dir.root}/demoWith2ledgers.yml.gz"/>
     <call target="stopPetraServer"/>

