--- a/csharp/ICT/Petra/Server/lib/MFinance/Common/Common.Posting.cs
+++ b/csharp/ICT/Petra/Server/lib/MFinance/Common/Common.Posting.cs
@@ -53,7 +53,7 @@ namespace Ict.Petra.Server.MFinance.Common
         /// <summary>
         /// This will be setup by CallForwarding, to allow me to call the FastReportsWrapper from here.
         /// </summary>
-        public static PrintReportOnClient PrintReportOnClientDelegate;
+        public static PrintReportOnClient PrintReportOnClientDelegate = null;
 
         private const int POSTING_LOGLEVEL = 1;
 
@@ -1362,7 +1362,10 @@ namespace Ict.Petra.Server.MFinance.Common
                         "param_ledger_name=\"" + LedgerName + "\""
                     };
                     String ParamStr = String.Join(",", Params);
-                    PrintReportOnClientDelegate("Batch Posting Register", ParamStr);
+                    if (PrintReportOnClientDelegate != null)
+                    {
+                        PrintReportOnClientDelegate("Batch Posting Register", ParamStr);
+                    }
                 }
             }

